app-creator/src/templates/templates-database.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ *  @fileoverview This file acts as a mock database storing our templates.
+ */
+
+interface TemplateItem {
+  id: string;
+  name: string;
+  variant: string;
+  template: {
+    [key: string]: {
+      name: string;
+      imageUrl: string;
+      template: string;
+    };
+  };
+}
+
+const legendTemplate = {
+  id: 'legend',
+  name: 'Legend',
+  'panel-template-0': {
+    id: 'panel-template-0',
+    editable: false,
+    hasDropzone: false,
+    children: ['map-template-0', 'panel-template-1'],
+    uniqueAttributes: {
+      layout: 'row',
+    },
+    style: {
+      height: '100%',
+      width: '100%',
+      padding: '0px',
+      margin: '0px',
+      position: 'relative',
+    },
+  },
+  'panel-template-1': {
+    id: 'panel-template-1',
+    editable: true,
+    hasDropzone: true,
+    children: [],
+    uniqueAttributes: {
+      layout: 'column',
+    },
+    style: {
+      height: '30%',
+      width: 'calc(100% - 32px)',
+      maxWidth: '500px',
+      padding: '0px',
+      margin: '0px',
+      color: 'black',
+      backgroundColor: '#FFFFFF',
+      borderWidth: '0px',
+      borderStyle: 'solid',
+      borderColor: 'black',
+      fontSize: '12px',
+      fontWeight: '300',
+      fontFamily: 'Roboto',
+      textAlign: 'left',
+      whiteSpace: 'normal',
+      shown: 'true',
+      position: 'absolute',
+      bottom: '16px',
+      left: '16px',
+    },
+  },
+  'map-template-0': {
+    id: 'map-template-0',
+    children: [],
+    uniqueAttributes: {
+      zoom: 4,
+    },
+    style: {
+      height: '100%',
+      width: '100%',
+    },
+  },
+};
+
+/**
+ * Placeholder database. This data will be stored on Datastore and will be fetched from an API endpoint.
+ */
+
+export const database: TemplateItem[] = [
+  {
+    id: 'side-panel',
+    name: 'Side Panel',
+    variant: 'left',
+    template: {
```

Comment:
Here I  thought about storing a state for each configuration. In our side panel example, we can either have a left panel layout or a right panel layout. So I'm storing a template string for each. The only difference between these two configurations is the ordering in the main panel `left: ["panel-template-1","map-template-0"]` `right: ["map-template-0", "panel-template-1"]`.

| Pros                                                      | Cons                                                                                       |
|-----------------------------------------------------------|--------------------------------------------------------------------------------------------|
| Extensible, any template can have multiple configurations. | We have to store the whole template state even though the difference can be small. (Can we store a diff instead?)     |
| Easy to manage.                  | Changing between configuration will be like changing templates so the data will be cleared. |